core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceModeTransformationManager.java

```diff
@@ -51,6 +57,8 @@ public class RudderDeviceModeTransformationManager {
 
     // checking how many seconds passed since last successful transformation
     private int deviceModeSleepCount = 0;
+    private int retryCount = 0;
+    private final Map<Integer, RudderTransformationRequest> messageMap = new HashMap<>();
```

Comment:
Rename to messageIdTransformationRequestMap
Ensure there is no race condition while accessing it.

Reply:
Rename is Done.
Regarding race condition: the operation on the `messageMap` is synchronous.